Lab2/server/nested_templates/userinterface.yaml

```diff
@@ -5,14 +5,24 @@ AWSTemplateFormatVersion: "2010-09-09"
 Transform: AWS::Serverless-2016-10-31
 Description: >
   Template to deploy cloudfront and s3 bucket for UI code
+Parameters:
+  IsCloudFrontAndS3PreProvisioned:
+    Type: String
+    Default: false    
+    Description: "Tells if cloudfront and s3 buckets are pre-provisioned or not. 
+    They get pre-provisioned when the workshop is running as a part of AWS Event through AWS event engine tool."
+Conditions:
+  IsNotRunningInEventEngine: !Not [ !Equals [ !Ref IsCloudFrontAndS3PreProvisioned, true] ]     
```

Comment:
why we need this variable, why can't we use the IsCloudFrontAndS3PreProvisioned?

Reply:
We need to provision resources in this template new IsCloudFrontAndS3PreProvisioned=false but we need true value for the Condition construct to provision resources. So trying to get the negate value of "IsCloudFrontAndS3PreProvisioned" here. IMO I feel this is bit easy to read.

Another way would be we can change the parameter name to "IsCloudFrontAndS3**Not**PreProvisioned" and can use it directly. In this case we also need to make similar change in parent stack and in deployment script so that it is uniform.